server/src/handlers/dataset_handler.rs

```diff
@@ -107,6 +115,8 @@ pub struct CreateDatasetRequest {
 pub async fn create_dataset(
     data: web::Json<CreateDatasetRequest>,
     pool: web::Data<Pool>,
+    redis_pool: web::Data<RedisPool>,
+    dataset_org_plan_sub: DatasetAndOrgWithSubAndPlan,
```

Comment:
This doesn't work since we can't have a `TR-Dataset` header before the dataset is created